flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1385,6 +1385,7 @@ private CompletedCheckpoint addCompletedCheckpointToStoreAndSubsumeOldest(
                 checkpointsCleaner.cleanCheckpointOnFailedStoring(completedCheckpoint, executor);
             }
 
+            reportFailedCheckpoint(completedCheckpoint, exception);
```

Comment:
I think it's not perfect that we are reporting failures and successful completions in two very different places. Can not we bring both reporting closer together? For example add the successful completion reporting to the `addCompletedCheckpointToStoreAndSubsumeOldest` method as well?

Reply:
I agree that it is better to keep both 'success' and 'fail' reporting in one place. It is why I wanted to move it to `completePendingCheckpoint` but it is not possible since `finalizeCheckpoint` can be called independently so 'fail' reporting should be inside of `finalizeCheckpoint`. So right now we report about the fail from `finalizeCheckpoint` and `addCompletedCheckpointToStoreAndSubsumeOldest` and also we report about 'success' from `completePendingCheckpoint`. I don't really like it but I don't think that moving 'succes' reporting to `addCompletedCheckpointToStoreAndSubsumeOldest` somehow improve the situation since the reporting should be the last step in the finalization of the checkpoint but moving this step into the method which is the last for now but can be changed in future seems not so promising